staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -183,6 +185,19 @@ func (f *ConfigFlags) toRawKubeConfigLoader() clientcmd.ClientConfig {
 	if f.Impersonate != nil {
 		overrides.AuthInfo.Impersonate = *f.Impersonate
 	}
+	if f.ImpersonateUserExtra != nil && len(*f.ImpersonateUserExtra) > 0 {
+		userExtras := make(map[string][]string)
+		for _, extra := range *f.ImpersonateUserExtra {
+			parts := strings.SplitN(extra, "=", 2)
```

Comment:
This does mean that keys cannot contain `=`, which I think is an acceptable restriction based on the stricter validation we do in newer parts of the codebase:

https://github.com/kubernetes/kubernetes/blob/4d0d412686ff3a40a50e0f0c36ba248e596222ce/staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go#L373-L380